lib/compiler.js

```diff
@@ -65,6 +65,13 @@ exports.compile = function (code, options) {
     });
 
     importExportVisitor.finalizeHoisting();
+
+    if (getOption(options, "wrapWithRun")) {
+      result.code = "module.run(function(){" +
+        magicString.toString() + "//*/\n});";
+
+      return result;
+    }
```

Comment:
> I think this should be whatever it is in the REPL (if you type this and hit return), so forcing it to be undefined seems undesirable there.

By opting into reify you're enabling esm in the repl so the executed code should be esm. Switching parse goals depending on code in the repl after opting in seems a bit odd.